src/backend/app/projects/project_schemas.py

```diff
@@ -80,6 +78,17 @@ class TaskOut(BaseModel):
     id: uuid.UUID
     project_task_index: int
     outline: Any = Field(exclude=True)
+    state: Optional[State] = None
+    contributor: Optional[str] = None
+
+    @validator("state", pre=True, always=True)
```

Comment:
How do you get the State ?
state of the tasks are in `task_events`  table.

Reply:
@nrjadkry  To get the state, I am joining the tasks table with the task_events table, where the state of the tasks is stored.

```query = """
    SELECT
        tasks.id AS id,
        tasks.project_task_index AS project_task_index,
        tasks.outline AS outline,
        task_events.state AS state,
        users.name AS contributor
    FROM tasks
    LEFT JOIN task_events ON tasks.id = task_events.task_id
    LEFT JOIN users ON task_events.user_id = users.id
    WHERE tasks.project_id = :project_id;
"""
```